providers/implementations/kem/ml_kem.c

```diff
@@ -126,8 +126,8 @@ static int mlkem_set_ctx_params(void *vctx, const OSSL_PARAM params[])
     if (params == NULL)
         return 1;
 
-    if ((p = OSSL_PARAM_locate_const(params, OSSL_KEM_PARAM_MLKEM_ENC_ENTROPY)) != NULL
-        && (p->data_size != MLKEM_ENCAP_ENTROPY
+    if ((p = OSSL_PARAM_locate_const(params, OSSL_KEM_PARAM_IKME)) != NULL
+        && (p->data_size < MLKEM_ENCAP_ENTROPY
             || (ctx->entropy = OPENSSL_memdup(p->data, MLKEM_ENCAP_ENTROPY)) == NULL))
         return 0;
```

Comment:
This entire file goes away in the upcoming refactor. So it likely best if you rebase on the ML-KEM refactor PR.

Reply:
Okay.  CI failures are relevant but don't reproduce locally.

Reply:
Might it be easier for you to pull the commits here into the refactor branch?
Otherwise, we're going to encounter an ordering conflict which serialises PRs.

Reply:
Which CI failures?  As for the conflict, I am happy to pull your changes in, but this PR will become obsolete and should be closed. Works for you?